### PR TITLE
HID: Stub IHidServer: 134 (SetNpadAnalogStickUseCenterClamp)

### PR DESCRIPTION
**Pull request by Tsubasa0504**
Allows "Power Pro" series to boot without Ignore Missing Services.

Fixes https://github.com/Ryujinx/Ryujinx-Games-List/issues/4930 https://github.com/Ryujinx/Ryujinx-Games-List/issues/4413
Close https://github.com/Ryujinx/Ryujinx/issues/1891

### CODE REVIEW

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 1]
```diff
@@ -1106,6 +1107,17 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
...
+	    //SetNpadUseAnalogStickUseCenterClamp(bool Enable, nn::applet::AppletResourceUserId)  
+	    public ResultCode SetNpadUseAnalogStickUseCenterClamp(ServiceCtx context)
+	    {
+	        _NpadAnalogStickCenterClampEnabled = context.RequestData.ReadBoolean();
+	        long appletResourceUserId = context.RequestData.ReadInt64();
+
+            Logger.Stub?.PrintStub(LogClass.ServiceHid, new { appletResourceUserId, _NpadAnalogStickCenterClampEnabled });
+
+            return ResultCode.Success;
+        }
```
Comment (agatemosu) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1120]:
```suggestion
        [CommandCmif(134)] //6.1.0+
        //SetNpadUseAnalogStickUseCenterClamp(bool Enable, nn::applet::AppletResourceUserId)
        public ResultCode SetNpadUseAnalogStickUseCenterClamp(ServiceCtx context)
        {
            _NpadAnalogStickCenterClampEnabled = context.RequestData.ReadBoolean();
            long appletResourceUserId = context.RequestData.ReadInt64();

            Logger.Stub?.PrintStub(LogClass.ServiceHid, new { appletResourceUserId, _NpadAnalogStickCenterClampEnabled });

            return ResultCode.Success;
    }
```

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ class IHidServer : IpcService
 
         private bool _sixAxisSensorFusionEnabled;
         private bool _unintendedHomeButtonInputProtectionEnabled;
+	    private bool _NpadAnalogStickCenterClampEnabled;
```
Comment (agatemosu) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:25]:
```suggestion
        private bool _NpadAnalogStickCenterClampEnabled;
```

**Review by agatemosu — COMMENTED** [review]
I help you format lol

**Comment by Tsubasa0504** [comment]
thanks

**Comment by MutantAura** [comment]
For future reference you can run `dotnet format` from the command line.

**Comment by GamerzHell9137** [comment]
Fixes https://github.com/Ryujinx/Ryujinx/issues/1891

**Comment by GamerzHell9137** [comment]
Power Pros is launching without ignoring services

image

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 3]
```diff
@@ -22,6 +22,7 @@ class IHidServer : IpcService
 
         private bool _sixAxisSensorFusionEnabled;
         private bool _unintendedHomeButtonInputProtectionEnabled;
+        private bool _NpadAnalogStickCenterClampEnabled;
```
Comment (gdkchan) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:25]:
```suggestion
        private bool _npadAnalogStickCenterClampEnabled;
```

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 4]
```diff
@@ -1106,6 +1107,17 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
             //       If one is found, it returns the npadIdType of the other Npad and a bool.
             //       If not, it returns nothing.
         }
+        [CommandCmif(134)] //6.1.0+
```
Comment (gdkchan) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1110]:
```suggestion
        [CommandCmif(134)] // 6.1.0+
```

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 5]
```diff
@@ -1106,6 +1107,17 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
...
+        //SetNpadUseAnalogStickUseCenterClamp(bool Enable, nn::applet::AppletResourceUserId)
```
Comment (gdkchan) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1111]:
```suggestion
        // SetNpadUseAnalogStickUseCenterClamp(bool Enable, nn::applet::AppletResourceUserId)
```

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 6]
```diff
@@ -1106,6 +1107,17 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
...
+        public ResultCode SetNpadUseAnalogStickUseCenterClamp(ServiceCtx context)
+        {
+            _NpadAnalogStickCenterClampEnabled = context.RequestData.ReadBoolean();
```
Comment (gdkchan) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1114]: Are you sure this is correct? `ReadBoolean` will only read 1 byte, there might be some alignment going on. It's worth checking if the applet resource user ID value is correct.

**Comment by AcK77** [comment]
It seems to pass a PID too, so right now the value of the bool may be wrong.

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 7]
```diff
@@ -1106,6 +1107,18 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
             //       If one is found, it returns the npadIdType of the other Npad and a bool.
             //       If not, it returns nothing.
         }
+        [CommandCmif(134)] // 6.1.0+
+        // SetNpadUseAnalogStickUseCenterClamp(bool Enable, nn::applet::AppletResourceUserId)
+        public ResultCode SetNpadUseAnalogStickUseCenterClamp(ServiceCtx context)
+        {
+            ulong PID = context.RequestData.ReadUInt64();
```
Comment (AcK77) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1114]:
```suggestion
            ulong pid = context.RequestData.ReadUInt64();
```

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 8]
```diff
@@ -1106,6 +1107,18 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
...
+            _NpadAnalogStickCenterClampEnabled = context.RequestData.ReadUInt32() != 0;
```
Comment (AcK77) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1115]:
```suggestion
            _npadAnalogStickCenterClampEnabled = context.RequestData.ReadUInt32() != 0;
```

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 9]
```diff
@@ -1106,6 +1107,18 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
...
+            long appletResourceUserId = context.RequestData.ReadInt64();
+
+            Logger.Stub?.PrintStub(LogClass.ServiceHid, new { appletResourceUserId, _NpadAnalogStickCenterClampEnabled });
```
Comment (AcK77) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1118]:
```suggestion
            Logger.Stub?.PrintStub(LogClass.ServiceHid, new { appletResourceUserId, _npadAnalogStickCenterClampEnabled });
```

**Review by AcK77 — APPROVED** [review]

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 10]
```diff
@@ -1106,6 +1107,18 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
             //       If one is found, it returns the npadIdType of the other Npad and a bool.
             //       If not, it returns nothing.
         }
+        [CommandCmif(134)] // 6.1.0+
```
Comment (makigumo) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1110]: Probably missing newline.
```suggestion

        [CommandCmif(134)] // 6.1.0+
```

File: src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs [hunk 11]
```diff
@@ -1106,6 +1107,18 @@ private void SetNpadJoyAssignmentModeSingleWithDestinationImpl(ServiceCtx contex
...
+        // SetNpadUseAnalogStickUseCenterClamp(bool Enable, nn::applet::AppletResourceUserId)
+        public ResultCode SetNpadUseAnalogStickUseCenterClamp(ServiceCtx context)
+        {
+            ulong pid = context.RequestData.ReadUInt64();
+            _npadAnalogStickCenterClampEnabled = context.RequestData.ReadUInt32() != 0;
+            long appletResourceUserId = context.RequestData.ReadInt64();
+
+            Logger.Stub?.PrintStub(LogClass.ServiceHid, new { appletResourceUserId, _npadAnalogStickCenterClampEnabled });
```
Comment (makigumo) [inline src/Ryujinx.HLE/HOS/Services/Hid/IHidServer.cs:1118]: Should probably also log the `pid`.
```suggestion
            Logger.Stub?.PrintStub(LogClass.ServiceHid, new { pid, appletResourceUserId, _npadAnalogStickCenterClampEnabled });
```

**Comment by Tsubasa0504** [comment]
bump

**Pull request merged**